common/leica_sep.c

```diff
@@ -63,27 +104,51 @@ static int leica_sep_transfer(struct leica_sep_funcs *f, char *cmd, char *data,
 	if (!command_received)
 		return -1;
 
+	if(leica_get_sep_payload(data, bytes_received))
+		return -1;
+
 	return bytes_received;
 }
 
-int leica_sep_get_board_id(struct leica_sep_funcs *f, struct leica_sep_board_id *board_id)
+/*
+ * leica_get_sep
+ *
+ * Function to communicate with the MCU and get parsed data from the SEP
+ * payload. The result is stored in an environment variable named <sep_name>.
+ * Available sep names are: 'board_id', 'production_flag'.
+ */
+int leica_get_sep(char *sep_name, int sep_name_length)
```

Comment:
I don't see any reason to pass a string here...
We provide an abstraction in the header file - this can also be an enumeration, which is easy to compare and does not require additional handling of the size.
As the output needs to be communicated over environment (`env_set`) this shoul be done in the command and not in the function (it can be called from anywhere...).